File: Practice/Week2/Week2.js
function calculate (min,max,step) {
    var sum =0;
    var a = min;
    var b = max;
    var arr = [];
    for(i = min; i<=max; i += step){
        sum += i;
        arr.push(i);
        b = i;
    }
    console.log(sum)
    return sum
}

calculate(1,3,1);
calculate(4, 8, 2);
calculate(-1, 2, 2);

var data ={
    "employees":[
    {"name":"John",
    "salary":30000,
    "manager":false },
    {"name":"Bob",
    "salary":60000,
    "manager":true },
    {"name":"Jenny",
    "salary":50000,
    "manager":false },
    {"name":"Tony",
    "salary":40000,
    "manager":false }]
    };
    
    var x = data.employees[0].manager;
    function avg (data) {
    var n = data.employees.length;
    var salary = 0;
    var count = 0;
        for(let i=0; i<n; i++) {
            if(data.employees[i].manager == false) {
                salary += data.employees[i].salary;
                count += 1;
            }
        }
    console.log(salary/count)
    }
    
avg(data);

function func(a) {
 var sum = 0;
    return (b,c) => {
        sum = a+(b*c);
        console.log(sum);
        return sum
    }

}

function  add(a,b) {
    
}

func(2)(3,4)
func(5)(1, -5)
func(-3)(2, 9)

function maxProduct (arr) {
    var n = arr.length;
    if(n<2) {
        console.log("No pairs exists"+"<br>");
        return;
    }
    // if(n==2) {
    //     document.write("the result is "+ arr[0]*arr[1])
    // }
    var result = arr[0]*arr[1];
    for(let i = 0; i<n ; i++) {
        for(let j = i+1 ; j < n; j++) {
            if(arr[i]*arr[j] > result)  {
                // var result2 = "";
                result = arr[i]*arr[j];
                // result2=Math.max(result,result2);
            }
        }
    }
    console.log(result)
}


maxProduct([5, 20, 2, 6]);
maxProduct([10, -20, 0, 3]);
maxProduct([10, -20, 0, -3]);
maxProduct([-1, 2]);
maxProduct([-1, 0, 2]);
maxProduct([5, -1, -2, 0]);
maxProduct([-5, -2]);

function twoSum(nums,target) {
    var n = nums.length;
    var a = nums[0];
    var b = nums[1];
    var x= 0;
    var y= 1;
    var final = nums[0]+nums[1];
    if(n<2) {
        document.write("Not enough elements for this function");
    }
    for(let i = 0; i < n; i++) {
        for(let j = i+1; j < n; j++) {
            if(nums[i] + nums[j] == target) {
                a = nums[i];
                b = nums[j];
                x = i;
                y = j;
                final = nums[i]+nums[j];
            }    
        }
    } return ("["+x+", "+y+"]")
}

let result=twoSum([2, 11, 7, 15], 9);
console.log(result);



function maxZeros(nums) {
    var count = 0;
    var result = 0;
    for(let i=0; i<nums.length; i++)
        if (nums[i] == 1) {
            count = 0;
        }
        else if (nums[i] == 0) {
            count += 1;
            result = Math.max(result,count);
        }
    console.log(result);
}

maxZeros([0, 1, 0, 0]);
maxZeros([1, 0, 0, 0, 0, 1, 0, 1, 0, 0]);
maxZeros([1, 1, 1, 1, 1]);
maxZeros([0, 0, 0, 1, 1]);

File: Practice/Week2/Week2.py
from calendar import c
from unittest import result


def calculate (min, max, step):
    sum = 0
    while min<=max :
        sum+=min
        min+=step
    print(sum)
    return sum


calculate(1, 3, 1)
calculate(4, 8, 2)
calculate(-1, 2, 2)

data = {
    "employees": [
        {"name":"John",
        "salary":30000,
        "manager":False },
        {"name":"Bob",
        "salary":60000,
        "manager":True },
        {"name":"Jenny",
        "salary":50000,
        "manager":False },
        {"name":"Tony",
        "salary":40000,
        "manager":False }
    ]
}

def avg(data):
    n = len(data["employees"])
    sum = 0
    count = 0
    for employee in data["employees"]:
        if not employee["manager"]:
            sum = sum + employee["salary"]
            count += 1
    print(int(sum/count))       
    return int(sum/count)

avg(data)


def func(a):
    def multiply(b,c):
        print(a+b*c)
        return a+b*c
    return multiply

func(2)(3, 4)
func(5)(1, -5)
func(-3)(2, 9)


def maxProduct(nums):
    n = len(nums)
    
    if (n<2):
        print("No pairs exists")
        return
    a = nums[0]
    b = nums[1]
    for i in range(0,n):
        for j in range(i+1,n):
            if (nums[i] * nums[j] > a * b):
                a = nums[i]
                b = nums[j]
    print(a*b)

maxProduct([5, 20, 2, 6])
maxProduct([10, -20, 0, 3])
maxProduct([10, -20, 0, -3])
maxProduct([-1, 2])
maxProduct([-1, 0, 2])
maxProduct([5,-1, -2, 0])
maxProduct([-5, -2])


def twoSum(nums, target):
    a=nums[0]
    b=nums[1]
    a=0
    b=1
    n=len(nums)
    for i in range(0,n):
        for j in range(i+1, n):
            if nums[i]+nums[j] == target:
                a=i
                b=j
    result = f"[{a},{b}]"
    return result


result = twoSum([2,11,7,15],9)
print(result)

def maxZeros(nums):
    count =0
    result =0
    n = len(nums)
    for i in range(0,n):
        if nums[i] == 1:
            count = 0
            continue
        if nums[i] == 0:
            count += 1
            result = max(count,result)
    print(result)
    return result


maxZeros([0, 1, 0, 0])
maxZeros([1, 0, 0, 0, 0, 1, 0, 1, 0, 0])
maxZeros([1, 1, 1, 1, 1]) 
maxZeros([0, 0, 0, 1, 1]) 